### ISSUE TITLE
Train-dev-test splits

### ISSUE THREAD

**Issue by Mahhos** [ThilinaRajapakse/simpletransformers#282]
Hi. Is there any way to use 3 datasets of train, dev, test for training, calculating hyperparameters, and evaluating the model? It seems that the code only uses the train set for training, and test set for evaluating the model. I need to use a separate validation set for computing hyperparameters and loss function. In the end, I will do prediction on an unlabeled dataset. So, I have 4 datasets (train-dev-test labeled, and one unlabeled dataset).

**Comment by ThilinaRajapakse** [comment]
You can think of train/eval/predict as analogous to train/dev/test. You can use those functions in any way you like to build your desired pipeline.

**Comment by Mahhos** [comment]
Would you please correct me if I am wrong?
1. I assume that the model performance (in current code) is reported on the eval-set (not on predict-set). However, I should calculate the performance on my test-set and only use the eval-set for fine-tuning the model (calculating hyperparameters, and loss function). Do you think I still can use the current pipeline? I just want to make sure that eval here has the same functionality as the validation function. 
2. You mean I can use the predict-set (as test-set), ignore the reported accuracy on eval-set, and manually measure the performance metrics (F-1, accuracy, etc) using the predicted labels on predict set?

**Comment by ThilinaRajapakse** [comment]
The library gives you three basic functions, `train`, `eval`, and `predict`. The data that you give to any of these functions is entirely up to you. The functionality of the three functions is as below.

1. `Train`: Trains the model on the given data.
2. `Eval`: Makes predictions on the given data and compares the predictions to the true labels (which should be in the data).
3. `Predict`: Makes predictions on the given data. Labels are not necessary and will be ignored even if they are provided.

So you could take your dataset and split it into 4 as train, validation, eval, and test (test is unlabelled, others are labelled). Then, your pipeline would look something like this.

1. `Train` on train data.
2. `Eval` on validation data (which yields performance metrics).
3. Repeat steps 1 and 2 while performing hyperparameters until you are happy with the results.
4. `Eval` on your evaluation data to confirm that the hyperparameter tuning was successful and will generalize to the data. This will also give you the metrics you need. Ideally, these scores should be comparable to the scores obtained in step 2.
5. `Predict` on your unlabelled test dataset.

**Comment by Mahhos** [comment]
Thank you so much. Just one question, by step 3 do I need to change the code? I assume that this should be handled by step 2 (converging and using the highest checkpoint to report the performance on the validation data).

**Comment by ThilinaRajapakse** [comment]
You will need to write the code to _execute_ each of these steps. You will have to handle the hyperparameter tuning and selecting the best model based on the metrics you get.

**Comment by Mahhos** [comment]
thank you.
